### PR TITLE
Remove imported token dialog

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Before users can remove an imported token they need to confirm this action. Here we provide a component that will be used for that confirmation.

# Changes

- New `ImportTokenRemoveConfirmation` component.
- Add `yesLabel` to the `ConfirmationModal` component.

# Tests

- PO and unit tests for the component.
- Available on [beta](https://beta.nns.ic0.app/) (design was changed to reuse the confirmation modal component).

pr version screenshot:
image

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary.

### CODE REVIEW

File: frontend/src/tests/lib/components/accounts/ImportTokenRemoveConfirmation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      nnsClose,
+    };
+  };
+
+  beforeEach(() => {
+    tokensStore.reset();
+    icrcCanistersStore.reset();
+    vi.restoreAllMocks();
+
+    page.mock({
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/accounts/ImportTokenRemoveConfirmation.spec.ts:42]: If you remove the dependency on `selectableUniversesStore`, don't forget to remove this as well.

File: frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  import type { Universe } from "$lib/types/universe";
+  import { nonNullish } from "@dfinity/utils";
+  import { selectableUniversesStore } from "$lib/derived/selectable-universes.derived";
+
+  export let ledgerCanisterId: Principal;
+
+  const dispatch = createEventDispatcher();
+
+  let universe: Universe | undefined;
+  $: universe = $selectableUniversesStore.find(
```
Comment (dskloetd) [inline frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte:16]: Isn't the universe the same as the `ledgerCanisterId`?
Regardless, I think it's better not to depend on the store here so if it's not the same, can we pass in an extra prop instead?

**Review by dskloetd — APPROVED** [review]

File: frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  testId="import-token-remove-confirmation-component"
+  disablePointerEvents={true}
+>
+  <svelte:fragment slot="title">
+    {$i18n.import_token.remove_confirmation_title}
+  </svelte:fragment>
+
+  <div class="content">
+    <div class="headline">
+      {#if nonNullish(universe)}<UniversePageSummary {universe} />{/if}
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte:33]: Agree. Will be done after this is merged.

File: frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      <Tag>{$i18n.import_token.imported_token}</Tag>
+    </div>
+    <p data-tid="description">
+      <Html text={$i18n.import_token.remove_confirmation_description} />
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte:37]: As I understand it, we should avoid using Html for content that can be entered by a user (like proposal text, inputs etc.). However, it’s more flexible to use HTML for text when we have full control over the content. Do you have a different opinion?

File: frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    </p>
+    <div class="toolbar">
+      <button
+        class="secondary"
+        data-tid="close-button"
+        on:click={() => dispatch("nnsClose")}
+      >
+        {$i18n.core.back}
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte:45]: Good catch!

File: frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte (hunk 4 above)
[the remark this answers is not in the document]
Comment (dskloetd) [reply frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte:37]: In my opinion we should only use `Html` when there is no other option.
It's easy to change things later and forget that you're using `Html` or just for things to go wrong because you're not using a proper template.

File: frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+  const dispatch = createEventDispatcher();
+
+  let universe: Universe | undefined;
+  $: universe = $selectableUniversesStore.find(
+    ({ canisterId }) => canisterId === ledgerCanisterId.toText()
+  );
+</script>
+
+<Modal
```
[the remark this answers is not in the document]
Comment (dskloetd) [reply frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte:21]: What is different? Isn't it a matter of having a slot for the content?

File: frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte:16]: The universe contains more information than just the canister ID:
```
export interface Universe {
  canisterId: UniverseCanisterIdText;
  summary?: SnsSummary;
  title: string;
  logo: string;
}
```
Yes, we could pass only the universe parameter and handle the logic in the parent component.

File: frontend/src/tests/lib/components/accounts/ImportTokenRemoveConfirmation.spec.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/components/accounts/ImportTokenRemoveConfirmation.spec.ts]: Thanks, done

File: frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte (hunk 6 above)
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte:16]: Title, close button, grey area.

image

vs

image

[the remark this answers is not in the document]
Comment (dskloetd) [reply frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte:16]: Is UX aware that they made this confirmation dialog different from other confirmation dialogs?
Is it intentional?
Why?

File: frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte (hunk 4 above)
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte]: It has been split into two messages.

**Review by dskloetd — APPROVED** [review]

File: frontend/src/tests/lib/components/accounts/ImportTokenRemoveConfirmation.spec.ts [hunk 7]
```diff
@@ -15,54 +15,28 @@ describe("ImportTokenRemoveConfirmation", () => {
...
     expect(await po.getUniversePageSummaryPo().getLogoUrl()).toEqual(tokenLogo);
   });
 
   it("should render token name", async () => {
-    const { po } = renderComponent();
+    const po = renderComponent();
     expect(await po.getUniversePageSummaryPo().getTitle()).toEqual(tokenName);
   });
-
-  it("should dispatch events", async () => {
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/accounts/ImportTokenRemoveConfirmation.spec.ts]: Why did you remove this?

File: frontend/src/lib/modals/common/ConfirmationModal.svelte [hunk 8]
```diff
@@ -23,7 +25,8 @@
       data-tid="confirm-yes"
       disabled={$busy}
       class="primary"
-      on:click={() => dispatch("nnsConfirm")}>{$i18n.core.confirm_yes}</button
+      on:click={() => dispatch("nnsConfirm")}
+      >{yesLabel ?? $i18n.core.confirm_yes}</button
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/lib/modals/common/ConfirmationModal.svelte:29]: Good catch. Forgot about it. Added https://github.com/dfinity/nns-dapp/pull/5366/files#diff-acb7cc41e636cf575094ed7c29d17e1570bfbf9836e755e0032046f1d16ab612R21

File: frontend/src/tests/page-objects/ImportTokenRemoveConfirmation.page-object.ts [hunk 9]
```diff
@@ -1,8 +1,9 @@
-import { ModalPo } from "$tests/page-objects/Modal.page-object";
+import { ConfirmationModalPo } from "$tests/page-objects/ConfirmationModal.page-object";
 import { UniversePageSummaryPo } from "$tests/page-objects/UniversePageSummary.page-object";
+import { BasePageObject } from "$tests/page-objects/base.page-object";
 import type { PageObjectElement } from "$tests/types/page-object.types";
 
-export class ImportTokenRemoveConfirmationPo extends ModalPo {
+export class ImportTokenRemoveConfirmationPo extends BasePageObject {
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/tests/page-objects/ImportTokenRemoveConfirmation.page-object.ts:6]: Good point, done [here](https://github.com/dfinity/nns-dapp/pull/5366).

File: frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte [hunk 10]
```diff
@@ -1,63 +1,38 @@
...
->
-  <svelte:fragment slot="title">
-    {$i18n.import_token.remove_confirmation_title}
-  </svelte:fragment>
-
-  <div class="content">
-    <div class="headline">
-      {#if nonNullish(universe)}<UniversePageSummary {universe} />{/if}
-      <Tag>{$i18n.import_token.imported_token}</Tag>
+<TestIdWrapper testId="import-token-remove-confirmation-component">
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/lib/components/accounts/ImportTokenRemoveConfirmation.svelte:13]: Thanks, replaced [here](https://github.com/dfinity/nns-dapp/pull/5366).

File: frontend/src/tests/lib/components/accounts/ImportTokenRemoveConfirmation.spec.ts (hunk 7 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/components/accounts/ImportTokenRemoveConfirmation.spec.ts]: Was not sure it worth it. Has been restored - https://github.com/dfinity/nns-dapp/pull/5366
